graphql/schemabuilder/pagination.go

```diff
@@ -702,23 +756,45 @@ func (c *connectionContext) consumeTextFilters(sb *schemaBuilder, m *method, typ
 	c.FilterTextFields = make(map[string]*graphql.Field)
 
 	for name, fn := range m.TextFilterFuncs {
-		funcTyp := getFuncReturnType(fn)
+		if fn.BatchFilterFunc != nil {
+			batchFuncTyp := getFuncReturnType(fn.BatchFilterFunc)
+			if batchFuncTyp != typeofMap {
+				return fmt.Errorf("invalid text filter field %s: unsupported return type %v, must be a map[batch.Index]string", name, batchFuncTyp)
+			}
+		}
+		if fn.FilterFunc != nil {
+			funcTyp := getFuncReturnType(fn.FilterFunc)
+			if funcTyp != typeOfString {
+				return fmt.Errorf("invalid text filter field %s: unsupported return type %v, must be a string", name, funcTyp)
+			}
+		}
 
-		if funcTyp != typeOfString {
-			return fmt.Errorf("invalid text filter field %s: unsupported return type %v, must be a string", name, funcTyp)
+		var field *graphql.Field = nil
```

Comment:
don't need the `= nil` part here.

Reply:
It needs to be defined beforehand, so that I can correctly initialize them later depending on if its a batch/nonbatch/fallback filter, and then I can do the error checking after. Since the error checking is the same for all 3 cases, its cleaner to just define these variables at the top

Reply:
oh, yeah, but you can just do `var field *graphql.Field` it automatically defaults to nil